stable_diffusion/stable_diffusion/unet.py

```diff
@@ -106,10 +106,11 @@ def __init__(
     def __call__(self, x, encoder_x, attn_mask, encoder_attn_mask):
         # Save the input to add to the output
         input_x = x
+        dtype = x.dtype
 
         # Perform the input norm and projection
         B, H, W, C = x.shape
-        x = self.norm(x).reshape(B, -1, C)
+        x = self.norm(x.astype(mx.float32)).astype(dtype).reshape(B, -1, C)
```

Comment:
Note for LN in LLMS we downcast before the affine which works fine and avoids needing to cast the weights at each iteration.  May be a little optimization to take the default compiled layernorm from MLX LM and use it her.

Reply:
Out of curiosity, is there a reason we are not moving the shapeless compile inside `mlx.nn` ?

Reply:
The only reason is that it doesn't do the up casting for the reduction.

Reply:
I was thinking about adding it there ... if you like that more feel free / I can add it, I would be on board.